includes/admin/class-wc-rest-payments-webhook-controller.php

```diff
@@ -229,6 +232,36 @@ private function process_webhook_expired_authorization( $event_body ) {
 		WC_Payments_Utils::mark_payment_expired( $order );
 	}
 
+	/**
+	 * Process webhook for a successul payment intent.
+	 *
+	 * @param array $event_body The event that triggered the webhook.
+	 *
+	 * @throws Rest_Request_Exception           Required parameters not found.
+	 * @throws Invalid_Payment_Method_Exception When unable to resolve charge ID to order.
+	 */
+	private function process_webhook_payment_intent_succeeded( $event_body ) {
+		$event_data   = $this->read_rest_property( $event_body, 'data' );
+		$event_object = $this->read_rest_property( $event_data, 'object' );
+
+		$intent_id = $this->read_rest_property( $event_object, 'id' );
+
+		// Look up the order related to this charge.
+		$order = $this->wcpay_db->order_from_intent_id( $intent_id );
+		if ( ! $order ) {
+			throw new Invalid_Payment_Method_Exception(
```

Comment:
What's the desired behaviour if we don't find the order here? Currently the webhook will get re-tried 3 times by our server if we return an error code here. I suppose this could be an intermittent DB failure, so that might be good?

Reply:
Good question, I think if an order is not found, we shouldn't update any of its order statuses and the webhook should be ignored silently on the client's side. 

In this code, it throws an exception up and it should be caught by these lines:
https://github.com/Automattic/woocommerce-payments/blob/a3ab5554b59aba45829e2c20c2cf30da5889039c/includes/admin/class-wc-rest-payments-webhook-controller.php#L144-L145
  
It should result in a `500` `WP_REST_Response`. Will this 500 allow the server to retry again?

Reply:
Yeah, the server will re-try if it gets an error back from the site. I think that's fine in this context.